File: Measurement Studio/Examples/DotNET/v4.5/UI.WinForms/Graph/ErrorBands/CustomErrorData/cs/ExplicitErrorMode.cs

using System;
using NationalInstruments.UI;
using NationalInstruments.UI.WindowsForms;

namespace NationalInstruments.Examples.CustomErrorData
{

	public class ExplicitErrorMode : XYErrorDataMode
	{
        private readonly ScatterPlot plot;
        private double[] errorData;

		public ExplicitErrorMode(ScatterPlot plot)
		{
            this.plot = plot;
		}

		public override void GetErrorData(XYErrorDataArgs args, out double[] highErrorData, out double[] lowErrorData)
		{
            // Use whatever error data we were given when PlotXYWithError was called last.
            highErrorData = errorData;
            lowErrorData = errorData;
        }

        public void PlotXYWithError(double[] xData, double[] yData, double[] errorData)
        {
            this.errorData = errorData;

            plot.PlotXY(xData, yData);
        }

	}

}

File: Measurement Studio/Examples/DotNET/v4.5/UI.WinForms/Graph/ErrorBands/CustomErrorData/vb/ExplicitErrorMode.vb

Imports System
Imports NationalInstruments.UI
Imports NationalInstruments.UI.WindowsForms

Public Class ExplicitErrorMode
    Inherits XYErrorDataMode
    Private ReadOnly plot As ScatterPlot
    Private errorData() As Double

    Public Sub New(ByVal plot As ScatterPlot)
        Me.plot = plot
    End Sub

    Public Overrides Sub GetErrorData(ByVal args As XYErrorDataArgs, ByRef highErrorData() As Double, ByRef lowErrorData() As Double)
        ' Use whatever error data we were given when PlotXYWithError was called last.
        highErrorData = errorData
        lowErrorData = errorData
    End Sub

    Public Sub PlotXYWithError(ByVal xData() As Double, ByVal yData() As Double, ByVal errorData() As Double)
        Me.errorData = errorData

        plot.PlotXY(xData, yData)
    End Sub

End Class
